com.unity.netcode.gameobjects/Runtime/Messaging/MessagingSystem.cs

```diff
@@ -215,19 +226,10 @@ public void HandleMessage(in MessageHeader header, FastBufferReader reader, ulon
                 SystemOwner = m_Owner,
                 SenderId = senderId,
                 Timestamp = timestamp,
-                Header = header
+                Header = header,
+                SerializedHeaderSize = serializedHeaderSize,
             };
-            var type = m_ReverseTypeMap[header.MessageType];
```

Comment:
Is the moving of this code related to the original fix?

Reply:
Yes and no... one iteration of the fix did require this code to be moved... when I backtracked on that direction I realized the code still needed to be moved because as it is, it's dispatching profiling/metrics events twice for some messages. I can move it into a separate PR...

Reply:
Yes lets move that into a different PR